### PR TITLE
[DOMPurify] Add forceKeepAttr property to attribute hook event

### PR DESCRIPTION
**Pull request by NEllusion**
**TL;DR** Last year the `forceKeepAttr` flag was added to the `sanitizeAttributeHook` event. This PR adds the corresponding type definition. 

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
* The line in the source code can be found here: https://github.com/cure53/DOMPurify/blob/main/src/purify.js#L994
* The PR that introduced this change can be found here: https://github.com/cure53/DOMPurify/pull/385
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
* This PR adds missing types from a very old DOMPurify version (v2.0.8)

### CODE REVIEW

**Review by peterblazejewicz — APPROVED** [review]
LGMT!
@NicholasEllul thx!

**Comment by NicholasEllul** [comment]
Ready to merge
